Userland/Libraries/LibJS/Runtime/AsyncFromSyncIteratorPrototype.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2021, David Tuin <davidot@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PromiseReaction.h"
+#include <LibJS/Runtime/AsyncFromSyncIteratorPrototype.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/IteratorOperations.h>
+#include <LibJS/Runtime/Promise.h>
+#include <LibJS/Runtime/PromiseConstructor.h>
+
+namespace JS {
+
+AsyncFromSyncIteratorPrototype::AsyncFromSyncIteratorPrototype(GlobalObject& global_object)
+    : PrototypeObject(*global_object.async_iterator_prototype())
+{
+}
+
+void AsyncFromSyncIteratorPrototype::initialize(GlobalObject& global_object)
+{
+    auto& vm = global_object.vm();
+    Object::initialize(global_object);
+
+    u8 attr = Attribute::Writable | Attribute::Configurable;
+    define_native_function(vm.names.next, next, 1, attr);
+    define_native_function(vm.names.return_, return_, 1, attr);
+    define_native_function(vm.names.throw_, throw_, 1, attr);
+}
+
+AsyncFromSyncIteratorPrototype::~AsyncFromSyncIteratorPrototype()
+{
+}
+
+// 27.2.1.1.1 IfAbruptRejectPromise ( value, capability ), https://tc39.es/ecma262/#sec-ifabruptrejectpromise
+#define TRY_OR_REJECT(vm, expression, capability)                                                         \
```

Comment:
This macro is copied from PromiseConstructor.cpp
Should we maybe make this a macro you can import? Or somehow wrap this in a normal function call?
(I happen to know we will need this in at least one other spot)

Reply:
Obviously we don't want to have it duplicated. Belongs in `PromiseReaction.h` next to `27.2.1.1 PromiseCapability Records`.

Reply:
...and then include that header where it's needed.